babeld/message.c

```diff
@@ -288,13 +288,18 @@ channels_len(unsigned char *channels)
 static int
 babel_packet_examin(const unsigned char *packet, int packetlen)
 {
-    unsigned i = 0, bodylen;
+    int i = 0, bodylen;
```

Comment:
why make these signed - why would they ever be < 0 ?

Reply:
First, it is sufficient to use a 32-bit int type as we only read two bytes into the variable. This also follows the existing code at Line 331 (see code below)

https://github.com/FRRouting/frr/blob/a89a78236cb03a2afa6dcc4227598964e6ca1a31/babeld/message.c#L324-L331

Second, if we still use unsigned, the compiler will warn that "comparison of integers of different signs" at Line 298.